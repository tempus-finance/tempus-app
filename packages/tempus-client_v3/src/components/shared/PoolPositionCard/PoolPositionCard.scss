.tc__poolPositionCard {
  display: flex;
  flex-direction: column;
  height: max-content;
  border: 1px solid var(--poolPositionCardBorder);
  width: 100%;
  background-color: var(--poolPositionCardBackground);
}

.tc__poolPositionCard-row {
  display: flex;
}

.tc__poolPositionCard-divider {
  width: 100%;
  height: 1px;
  background-color: var(--poolPositionCardBorder);
}

.tc__poolPositionCard-cell {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-grow: 1;
  padding: 16px;
  border: 1px solid var(--poolPositionCardBorder);

  &[data-size='small'] {
    padding: 8px;
  }
}

.tc__poolPositionCard-cellData {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0px 16px;
}

.tc__poolPositionCard-cellDataRow {
  display: flex;
  align-items: center;
  justify-content: center;
}

.tc__poolPositionCard-seeMore {
  flex-grow: 1;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}
