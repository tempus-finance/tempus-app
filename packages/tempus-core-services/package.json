{
  "name": "tempus-core-services",
  "version": "0.0.1",
  "private": true,
  "description": "Tempus core Services",
  "keywords": [
    "typescript",
    "boilerplate",
    "esbuild"
  ],
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "scripts": {
    "lint": "eslint src/ --ext .js,.jsx,.ts,.tsx",
    "test": "jest",
    "clean": "rm -rf dist build package",
    "ts-node": "ts-node",
    "docs": "typedoc --entryPoints src/index.ts",
    "build": "npm run clean && tsc -p tsconfig.json",
    "build-all": "npm run clean && npm run build && npm run esbuild-node && npm run esbuild-node",
    "esbuild-node": "esbuild src/index.ts --bundle --platform=node --minify --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:dev": "esbuild src/index.ts --bundle --platform=node --sourcemap=external --outfile=dist/esbuild/index.js",
    "esbuild-node:watch": "esbuild src/index.ts --bundle --platform=node --watch --sourcemap=external --outfile=dist/esbuild/index.js"
  },
  "dependencies": {
    "@ethersproject/providers": "^5.5.2",
    "@types/jest": "^24.9.1",
    "@types/node": "^12.20.15",
    "axios": "^0.22.0",
    "date-fns": "^2.28.0",
    "ethers": "^5.4.6"
  },
  "devDependencies": {
    "esbuild": "^0.11.11",
    "eslint": "^7.22.0",
    "jest": "^26.6.3",
    "ts-jest": "^26.5.4",
    "ts-node": "^9.1.1",
    "typedoc": "^0.20.35"
  },
  "peerDependencies": {
    "@typescript-eslint/eslint-plugin": "^4.28.1",
    "@typescript-eslint/parser": "^4.28.1",
    "eslint-config-airbnb-typescript": "^12.3.1"
  }
}
