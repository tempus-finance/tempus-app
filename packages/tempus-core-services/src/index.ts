import ERC20ABI from './abi/ERC20.json';
import VaultABI from './abi/Vault.json';

export { ERC20ABI, VaultABI };

export type { ERC20 } from './abi';

export type {
  AvailableToDeposit,
  Chain,
  ChainConfig,
  Config,
  DynamicTempusPool,
  PoolShares,
  ProtocolDisplayName,
  ProtocolName,
  Ticker,
  NativeTokenTicker,
  TempusPool,
  TokenPrecision,
  TokenTypePrecision,
  YearnData,
  YearnDataApy,
  YearnDataToken,
} from './interfaces';

export type {
  DepositedEvent,
  ERC20TokenService,
  PoolBalanceChangedEvent,
  PoolBalanceChangedEventListener,
  RedeemedEvent,
  TempusAMMJoinKind,
  TempusAMMExitKind,
  TempusAMMService,
  TempusControllerService,
  TempusPoolService,
  TransferEventListener,
  VaultService,
  StatisticsService,
  StorageService,
  SwapEvent,
} from './services';

export type { Numberish } from './datastructures';

export { Decimal, ZERO, ONE, DEFAULT_DECIMAL_PRECISION } from './datastructures';

export { SwapKind } from './services';

export {
  getChainlinkFeed,
  getCoingeckoRate,
  getDefaultProvider,
  getERC20TokenService,
  initServices,
  getServices,
  getDefinedServices,
} from './services';

export {
  NumberUtils,
  DecimalUtils,
  capitalize,
  chainIdToChainName,
  chainIdHexToChainName,
  chainNameToHexChainId,
  chainToTicker,
  decreasePrecision,
  div18f,
  getCookie,
  getLibrary,
  getPastDaysNumber,
  getProvider,
  getProviderFromSignerOrProvider,
  getRangeFrom,
  getTokenPrecision,
  increasePrecision,
  isZeroString,
  mul18f,
  prettifyChainName,
  prettifyChainNameLong,
  rayToDai,
  reverseString,
  shortenAccount,
  wadToDai,
  wait,
  prettifyProtocolName,
} from './utils';

export * from './constants';
