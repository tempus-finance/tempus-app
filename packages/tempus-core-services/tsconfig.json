{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,

    "importHelpers": true,
    "experimentalDecorators": true,
    "sourceMap": true,
    "outDir": "./dist/",
    "types": ["node", "jest"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["node_modules"]
}
