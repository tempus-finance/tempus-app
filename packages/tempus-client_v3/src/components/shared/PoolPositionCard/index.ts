import PoolPositionCard from './PoolPositionCard';

export default PoolPositionCard;
