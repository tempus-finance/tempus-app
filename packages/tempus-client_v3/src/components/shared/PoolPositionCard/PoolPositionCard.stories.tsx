import { ComponentStory, ComponentMeta } from '@storybook/react';
import { Decimal } from 'tempus-core-services';
import PoolPositionCard from './PoolPositionCard';

export default {
  title: 'PoolPositionCard',
  component: PoolPositionCard,
  argTypes: {
    term: {
      control: {
        type: 'date',
      },
    },
  },
} as ComponentMeta<typeof PoolPositionCard>;

const style = {
  display: 'flex',
  paddingTop: '50px',
  height: '300px',
  width: '800px',
  marginLeft: 'auto',
  marginRight: 'auto',
};

const Template: ComponentStory<typeof PoolPositionCard> = args => (
  <div style={style}>
    <PoolPositionCard {...args} />
  </div>
);

export const Primary = Template.bind({});
Primary.args = {
  apr: 0.1,
  term: new Date(2022, 2, 4),
  profitLoss: new Decimal(10.29651),
  balance: new Decimal(20.9294),
  totalYieldEarned: new Decimal(1.12345),
  projectedTotalYield: new Decimal(3.181215),
  tokenExchangeRate: new Decimal(2000),
  tokenDecimals: 4,
  tokenTicker: 'ETH',
};
