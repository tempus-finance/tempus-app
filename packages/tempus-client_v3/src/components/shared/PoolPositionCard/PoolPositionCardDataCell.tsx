import { FC, useMemo } from 'react';
import { Decimal, DecimalUtils, Ticker } from 'tempus-core-services';
import Typography from '../Typography';

interface PoolPositionCardDataCellProps {
  label: string;
  value: Decimal;
  tokenTicker: Ticker;
  tokenExchangeRate: Decimal;
  tokenDecimals: number;
}

const PoolPositionCardDataCell: FC<PoolPositionCardDataCellProps> = props => {
  const { label, value, tokenTicker, tokenExchangeRate, tokenDecimals } = props;

  const valueFormatted = useMemo(() => DecimalUtils.formatToCurrency(value, tokenDecimals), [value, tokenDecimals]);

  const valueFiatFormatted = useMemo(
    () => DecimalUtils.formatToCurrency(value.mul(tokenExchangeRate), 2, '$'),
    [value, tokenExchangeRate],
  );

  return (
    <div className="tc__poolPositionCard-cellData">
      <Typography variant="body-secondary" weight="bold" color="text-secondary">
        {label}
      </Typography>
      <div className="tc__poolPositionCard-cellDataRow">
        <Typography variant="body-primary" weight="medium" type="mono">
          {valueFormatted}
        </Typography>
        &nbsp; {/* space character */}
        <Typography variant="body-primary" weight="medium" color="text-secondary">
          {tokenTicker}
        </Typography>
      </div>
      <div className="tc__poolPositionCard-cellDataRow">
        <Typography variant="body-secondary" weight="medium" type="mono">
          {valueFiatFormatted}
        </Typography>
        &nbsp; {/* space character */}
        <Typography variant="body-secondary" weight="medium" color="text-secondary">
          USD
        </Typography>
      </div>
    </div>
  );
};
export default PoolPositionCardDataCell;
