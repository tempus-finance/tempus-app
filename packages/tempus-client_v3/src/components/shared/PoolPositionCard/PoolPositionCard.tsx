import { FC, useCallback, useMemo, useState } from 'react';
import { Decimal, DecimalUtils, Ticker } from 'tempus-core-services';
import ButtonWrapper from '../ButtonWrapper';
import FormattedDate from '../FormattedDate';
import Icon from '../Icon';
import Typography from '../Typography';
import PoolPositionCardDataCell from './PoolPositionCardDataCell';

import './PoolPositionCard.scss';

export interface PoolPositionCardProps {
  apr: number;
  term: Date;
  profitLoss: Decimal;
  balance: Decimal;
  totalYieldEarned: Decimal;
  projectedTotalYield: Decimal;
  // Used to convert and show above values in fiat currency
  tokenExchangeRate: Decimal;
  tokenDecimals: number;
  tokenTicker: Ticker;
}

const PoolPositionCard: FC<PoolPositionCardProps> = props => {
  const {
    apr,
    term,
    profitLoss,
    balance,
    totalYieldEarned,
    projectedTotalYield,
    tokenExchangeRate,
    tokenDecimals,
    tokenTicker,
  } = props;

  const [open, setOpen] = useState<boolean>(false);

  const onToggleClick = useCallback(() => {
    setOpen(prevState => !prevState);
  }, []);

  const aprFormatted = useMemo(() => DecimalUtils.formatPercentage(apr), [apr]);

  return (
    <div className="tc__poolPositionCard">
      <div className="tc__poolPositionCard-row">
        <div className="tc__poolPositionCard-cell">
          <div className="tc__poolPositionCard-cellData">
            <Typography variant="body-secondary" weight="bold" color="text-secondary">
              APR
            </Typography>
            <Typography variant="body-primary" weight="medium" type="mono">
              {aprFormatted}
            </Typography>
          </div>
        </div>
        <div className="tc__poolPositionCard-cell">
          <div className="tc__poolPositionCard-cellData">
            <Typography variant="body-secondary" weight="bold" color="text-secondary">
              Term
            </Typography>
            <FormattedDate date={term} size="medium" separatorContrast="high" />
          </div>
        </div>
      </div>
      {open && (
        <>
          <div className="tc__poolPositionCard-divider" />
          <div className="tc__poolPositionCard-row">
            <div className="tc__poolPositionCard-cell">
              <PoolPositionCardDataCell
                label="Profit/loss"
                tokenDecimals={tokenDecimals}
                tokenExchangeRate={tokenExchangeRate}
                tokenTicker={tokenTicker}
                value={profitLoss}
              />
              <PoolPositionCardDataCell
                label="Balance"
                tokenDecimals={tokenDecimals}
                tokenExchangeRate={tokenExchangeRate}
                tokenTicker={tokenTicker}
                value={balance}
              />
              <PoolPositionCardDataCell
                label="Total Yield Earned"
                tokenDecimals={tokenDecimals}
                tokenExchangeRate={tokenExchangeRate}
                tokenTicker={tokenTicker}
                value={totalYieldEarned}
              />
              <PoolPositionCardDataCell
                label="Projected Total Yield"
                tokenDecimals={tokenDecimals}
                tokenExchangeRate={tokenExchangeRate}
                tokenTicker={tokenTicker}
                value={projectedTotalYield}
              />
            </div>
          </div>
          <div className="tc__poolPositionCard-divider" />
        </>
      )}

      <div className="tc__poolPositionCard-row">
        <div className="tc__poolPositionCard-cell" data-size="small">
          <ButtonWrapper onClick={onToggleClick}>
            <div className="tc__poolPositionCard-seeMore">
              <Typography variant="body-primary" weight="bold">
                {open ? 'See less' : 'See more'}
              </Typography>
              &nbsp; {/* space character */}
              <Icon variant={open ? 'up-chevron' : 'down-chevron'} size="small" />
            </div>
          </ButtonWrapper>
        </div>
      </div>
    </div>
  );
};
export default PoolPositionCard;
