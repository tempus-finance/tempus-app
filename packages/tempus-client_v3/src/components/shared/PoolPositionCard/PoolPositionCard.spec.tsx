import { fireEvent, render } from '@testing-library/react';
import { Decimal } from 'tempus-core-services';
import I18nProvider from '../../../i18n/I18nProvider';
import PoolPositionCard, { PoolPositionCardProps } from './PoolPositionCard';

const defaultProps: PoolPositionCardProps = {
  apr: 0.1,
  term: new Date(2022, 2, 4),
  profitLoss: new Decimal(10.29651),
  balance: new Decimal(20.9294),
  totalYieldEarned: new Decimal(1.12345),
  projectedTotalYield: new Decimal(3.181215),
  tokenExchangeRate: new Decimal(2000),
  tokenDecimals: 4,
  tokenTicker: 'ETH',
};

const subject = (props: PoolPositionCardProps) =>
  render(
    <I18nProvider>
      <PoolPositionCard {...props} />
    </I18nProvider>,
  );

describe('WalletButton', () => {
  it('renders a collapsed pool position card', () => {
    const { container, getByText } = subject(defaultProps);

    const actualText = getByText('See more');
    expect(actualText).not.toBeNull();

    expect(container).toMatchSnapshot();
  });

  it('expands pool position card when user clicks on see more', () => {
    const { container, queryByRole, getByText } = subject(defaultProps);

    const actualButton = queryByRole('button');
    expect(actualButton).not.toBeNull();

    fireEvent.click(actualButton as Element);

    const actualText = getByText('See less');
    expect(actualText).not.toBeNull();

    expect(container).toMatchSnapshot();
  });
});
