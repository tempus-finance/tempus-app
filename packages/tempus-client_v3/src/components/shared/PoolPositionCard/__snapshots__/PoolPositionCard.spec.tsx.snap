// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoolPositionCard expands pool position card when user clicks on see more 1`] = `
<div>
  <div
    class="tc__poolPositionCard"
  >
    <div
      class="tc__poolPositionCard-row"
    >
      <div
        class="tc__poolPositionCard-cell"
      >
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            APR
          </div>
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
          >
            10.00%
          </div>
        </div>
      </div>
      <div
        class="tc__poolPositionCard-cell"
      >
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Term
          </div>
          <span
            class="tc__formatted-date"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              04
            </div>
            <span
              class="tc__formatted-date__separator tc__formatted-date__separator-high-contrast"
            />
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              Mar
            </div>
            <span
              class="tc__formatted-date__separator tc__formatted-date__separator-high-contrast"
            />
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              24
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      class="tc__poolPositionCard-divider"
    />
    <div
      class="tc__poolPositionCard-row"
    >
      <div
        class="tc__poolPositionCard-cell"
      >
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Profit/loss
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              10.2965
            </div>
              
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              ETH
            </div>
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              $20,593.02
            </div>
              
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              USD
            </div>
          </div>
        </div>
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Balance
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              20.9294
            </div>
              
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              ETH
            </div>
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              $41,858.80
            </div>
              
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              USD
            </div>
          </div>
        </div>
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Total Yield Earned
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              1.1234
            </div>
              
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              ETH
            </div>
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              $2,246.90
            </div>
              
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              USD
            </div>
          </div>
        </div>
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Projected Total Yield
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              3.1812
            </div>
              
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              ETH
            </div>
          </div>
          <div
            class="tc__poolPositionCard-cellDataRow"
          >
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              $6,362.43
            </div>
              
            <div
              style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              USD
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tc__poolPositionCard-divider"
    />
    <div
      class="tc__poolPositionCard-row"
    >
      <div
        class="tc__poolPositionCard-cell"
        data-size="small"
      >
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <div
            class="tc__poolPositionCard-seeMore"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              See less
            </div>
              
            <svg
              class="tc__icon tc__icon-up-chevron"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_909_5449)"
              >
                <path
                  d="M15.7372 11.7636C15.6546 11.8385 15.5564 11.898 15.4481 11.9385C15.3398 11.9791 15.2237 12 15.1064 12C14.9892 12 14.873 11.9791 14.7648 11.9385C14.6565 11.898 14.5583 11.8385 14.4757 11.7636L7.99924 5.9304L1.5228 11.7636C1.35551 11.9141 1.12862 11.9986 0.892036 11.9986C0.655453 11.9986 0.42856 11.9141 0.261271 11.7636C0.0939821 11.6132 0 11.4091 0 11.1963C0 10.9835 0.0939821 10.7794 0.261271 10.629L7.36848 4.23636C7.45107 4.16147 7.54932 4.10202 7.65758 4.06145C7.76584 4.02089 7.88196 4 7.99924 4C8.11652 4 8.23264 4.02089 8.3409 4.06145C8.44916 4.10202 8.54742 4.16147 8.63001 4.23636L15.7372 10.629C15.8205 10.7032 15.8866 10.7916 15.9317 10.889C15.9768 10.9864 16 11.0908 16 11.1963C16 11.3018 15.9768 11.4062 15.9317 11.5036C15.8866 11.601 15.8205 11.6894 15.7372 11.7636Z"
                  fill="#222222"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_909_5449"
                >
                  <rect
                    fill="white"
                    height="16"
                    width="16"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PoolPositionCard renders a collapsed pool position card 1`] = `
<div>
  <div
    class="tc__poolPositionCard"
  >
    <div
      class="tc__poolPositionCard-row"
    >
      <div
        class="tc__poolPositionCard-cell"
      >
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            APR
          </div>
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
          >
            10.00%
          </div>
        </div>
      </div>
      <div
        class="tc__poolPositionCard-cell"
      >
        <div
          class="tc__poolPositionCard-cellData"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(71, 67, 80); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Term
          </div>
          <span
            class="tc__formatted-date"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              04
            </div>
            <span
              class="tc__formatted-date__separator tc__formatted-date__separator-high-contrast"
            />
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              Mar
            </div>
            <span
              class="tc__formatted-date__separator tc__formatted-date__separator-high-contrast"
            />
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
            >
              24
            </div>
          </span>
        </div>
      </div>
    </div>
    <div
      class="tc__poolPositionCard-row"
    >
      <div
        class="tc__poolPositionCard-cell"
        data-size="small"
      >
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <div
            class="tc__poolPositionCard-seeMore"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              See more
            </div>
              
            <svg
              class="tc__icon tc__icon-down-chevron"
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_909_5439)"
              >
                <path
                  d="M15.7387 5.36992L8.63082 11.7636C8.54823 11.8385 8.44996 11.898 8.34169 11.9385C8.23342 11.9791 8.11729 12 8 12C7.88271 12 7.76658 11.9791 7.65831 11.9385C7.55004 11.898 7.45177 11.8385 7.36918 11.7636L0.261296 5.36992C0.0939908 5.21942 -2.49304e-09 5.01531 0 4.80248C2.49304e-09 4.58965 0.0939908 4.38554 0.261296 4.23504C0.428601 4.08455 0.655515 4 0.89212 4C1.12873 4 1.35564 4.08455 1.52294 4.23504L8 10.0693L14.4771 4.23504C14.6444 4.08455 14.8713 4 15.1079 4C15.3445 4 15.5714 4.08455 15.7387 4.23504C15.906 4.38554 16 4.58965 16 4.80248C16 5.01531 15.906 5.21942 15.7387 5.36992Z"
                  fill="#222222"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_909_5439"
                >
                  <rect
                    fill="white"
                    height="16"
                    width="16"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
